stunnel/stunnel.ml

```diff
@@ -42,154 +41,155 @@ let timeoutidle = ref None
+   * These are all safe in the presence of threads *)
+  type fd_operation =
     | Dup2 of Unix.file_descr * Unix.file_descr
     | Close of Unix.file_descr
-		  
+
   let do_fd_operation = function
-	| Dup2(a, b) -> Unix.dup2 a b
-	| Close a -> Unix.close a
+    | Dup2(a, b) -> Unix.dup2 a b
+    | Close a -> Unix.close a
 end
 
-type pid = 
+type pid =
   | StdFork of int (** we forked and exec'ed. This is the pid *)
   | FEFork of Forkhelpers.pidty (** the forkhelpers module did it for us. *)
   | Nopid
 
-let string_of_pid = function
+(* let string_of_pid = function
   | StdFork x -> Printf.sprintf "(StdFork %d)" x
   | FEFork x -> Forkhelpers.string_of_pidty x
-  | Nopid -> "None"
+  | Nopid -> "None" *)
 
 let getpid ty =
   match ty with
-    | StdFork pid -> pid
-    | FEFork pid -> Forkhelpers.getpid pid
-    | Nopid -> failwith "No pid!"
+  | StdFork pid -> pid
+  | FEFork pid -> Forkhelpers.getpid pid
+  | Nopid -> failwith "No pid!"
 
 type t = { mutable pid: pid; fd: Unix.file_descr; host: string; port: int; 
-	   connected_time: float;
-	   unique_id: int option;
-	   mutable logfile: string;
-	   verified: bool;
-	   legacy: bool;
-	 }
+           connected_time: float;
+           unique_id: int option;
+           mutable logfile: string;
+           verified: bool;
+           legacy: bool;
+         }
 
 let config_file verify_cert extended_diagnosis host port legacy =
 
-	let good_ciphers () = match !good_ciphersuites with
-		| None -> raise (Stunnel_error "good_ciphersuites is unset in the OCaml Stunnel module.")
-		| Some s -> s
-	in
-
-	let lines = [
-		"client=yes"; "foreground=yes"; "socket = r:TCP_NODELAY=1"; "socket = r:SO_KEEPALIVE=1"; "socket = a:SO_KEEPALIVE=1";
-		(match !timeoutidle with None -> "" | Some x -> Printf.sprintf "TIMEOUTidle = %d" x);
-		Printf.sprintf "connect=%s:%d" host port;
-		"fips = no"; (* stunnel fips-mode stops us using sslVersion other than TLSv1 which means 1.0 only. *)
-	] @	(if extended_diagnosis then
-			["debug=4"]
-		else
-			[]
-	) @ (
-		if verify_cert then
-			["verify=2";
-			sprintf "CApath=%s" certificate_path;
-			sprintf "CRLpath=%s" crl_path]
-		else
-			[]
-	) @ (
-		if legacy then [
-			"sslVersion = all";
-			"options = NO_SSLv2";
-			"options = NO_SSLv3";
-			"ciphers = " ^ (good_ciphers ()) ^ (match !legacy_ciphersuites with "" -> "" | s -> (":" ^ s))
-		] else [
-			"sslVersion = TLSv1.2";
-			"ciphers = " ^ (good_ciphers ());
-		]
-	)
+  let good_ciphers () = match !good_ciphersuites with
+    | None -> raise (Stunnel_error "good_ciphersuites is unset in the OCaml Stunnel module.")
+    | Some s -> s
   in
-    String.concat "" (List.map (fun x -> x ^ "\n") lines)
+
+  let lines = [
+    "client=yes"; "foreground=yes"; "socket = r:TCP_NODELAY=1"; "socket = r:SO_KEEPALIVE=1"; "socket = a:SO_KEEPALIVE=1";
+    (match !timeoutidle with None -> "" | Some x -> Printf.sprintf "TIMEOUTidle = %d" x);
+    Printf.sprintf "connect=%s:%d" host port;
```

Comment:
This would be better written using `List.concat` but maybe not now.

Reply:
Yes, but this is reindenting only for now. If somebody has time, this module could appreciate some cleanup